canopen_fake_slaves/include/canopen_fake_slaves/basic_slave.hpp

```diff
@@ -49,6 +57,21 @@ class SimpleSlave : public canopen::BasicSlave
     uint32_t val = (*this)[idx][subidx];
     (*this)[0x4001][0] = val;
     this->TpdoEvent(0);
+    // Publish periodic message
+    message_thread = std::thread(std::bind(&SimpleSlave::fake_periodic_messages, this));
```

Comment:
We should not create a new tread each time. For now, a small `if` should help. An idea would be:

```suggestion
    if not message_thread:
        message_thread = std::thread(std::bind(&SimpleSlave::fake_periodic_messages, this));
```

Also, what I am not sure. You are using here class member, but the method does not have self. We should check `global` or similar attribute for the variable should be used.